sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_resource_configuration.py

```diff
@@ -31,7 +33,59 @@ def _validate(self):
             raise ValidationException(
                 message=msg,
                 no_personal_data_message=msg,
-                target=ErrorTarget.JOB,
+                target=ErrorTarget.SPARK_JOB,
+                error_category=ErrorCategory.USER_ERROR,
+            )
+        if self.instance_type.lower() not in self.instance_type_list:
+            msg = "Instance type must be specified for the list of {}".format(",".join(self.instance_type_list))
+            raise ValidationException(
+                message=msg,
+                no_personal_data_message=msg,
+                target=ErrorTarget.SPARK_JOB,
+                error_category=ErrorCategory.USER_ERROR,
+            )
+
+        # runtime_version type is either float or str
+        if isinstance(self.runtime_version, float):
+            if self.runtime_version < 3.1 or self.runtime_version >= 3.3:
+                msg = "runtime version should be either 3.1 or 3.2"
+                raise ValidationException(
+                    message=msg,
+                    no_personal_data_message=msg,
+                    target=ErrorTarget.SPARK_JOB,
+                    error_category=ErrorCategory.USER_ERROR,
+                )
+        elif isinstance(self.runtime_version, str):
+            runtime_arr = self.runtime_version.split('.')
+            try:
+                for runtime in runtime_arr:
+                    int(runtime)
+            except ValueError:
+                raise ValueError("runtime_version should only contain numbers")
+            if len(runtime_arr) <= 1:
```

Comment:
why not just compare the str with '3.1' and '3.2‘？

Reply:
if passing `3.1.abc` as a string to the backend, backend would throw error as well, since they are using C# function to transfer the string, so I have to make sure each number between a dot is a real number